apps/settings/js/panels/addon_details/panel.js

```diff
@@ -31,14 +31,23 @@ define(function(require) {
       },
 
       onBeforeShow: function(panel, options) {
-        this._curAddon = options.addon;
+        return Promise.resolve((() => {
+          if (options.addon) {
+            return options.addon;
+          } else if (options.manifestURL) {
+            return AddonManager.findAddonByManifestURL(options.manifestURL);
+          }
```

Comment:
Why we have to explicitly wrap this into a function ? Can't we just make sure what the app is at first and directly use that ?